components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/test/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerTest.java

```diff
@@ -437,6 +447,97 @@ public void testUpdateExistingSecretDescription() throws Exception {
                 "Existing id should be equal to the replaced id");
     }
 
+    @Test(priority = 28)
+    public void testAddIdpSecrets() throws Exception {
+
+        secretManager.addSecretType(getSampleSecretTypeAdd(
+                SecretConstants.IDN_SECRET_TYPE_IDP_SECRETS, "Secret type of IDP related secret properties"
+        ));
+        IdentityProvider identityProvider = buildIDPObject();
+        encryptSecret(SAMPLE_SECRET_VALUE1);
+        identityProviderSecretsProcessor.addOrUpdateIdpSecrets(identityProvider);
+        
+        for (String secretName : idpSecretReferences(identityProvider)) {
+            assertTrue(
+                    secretManager.isSecretExist(SecretConstants.IDN_SECRET_TYPE_IDP_SECRETS, secretName),
+                    "Expected secret is not found in the data source.");
+        }
+    }
+
+    @Test(priority = 29)
+    public void testGetIdpSecrets() throws Exception {
+
+        secretManager.addSecretType(getSampleSecretTypeAdd(
+                SecretConstants.IDN_SECRET_TYPE_IDP_SECRETS, "Secret type of IDP related secret properties"
+        ));
+        IdentityProvider identityProvider = buildIDPObject();
+        encryptSecret(SAMPLE_SECRET_VALUE1);
+        IdentityProvider addedIdp = identityProviderSecretsProcessor.addOrUpdateIdpSecrets(identityProvider);
+
+        decryptSecret(ENCRYPTED_VALUE1);
+        IdentityProvider updatedIdp = identityProviderSecretsProcessor.getIdpSecrets(addedIdp);
+
+        for (Property property : updatedIdp.getFederatedAuthenticatorConfigs()[0].getProperties()) {
+            if (property.isConfidential()) {
+                assertEquals(SAMPLE_SECRET_VALUE1, property.getValue());
+            }
+        }
+
+//        Property[] idpProperties = identityProvider.getFederatedAuthenticatorConfigs()[0].getProperties();
```

Comment:
Shall we remove these lines?

Reply:
Resolved with a608462e19bdcdde4d7c1f8d83fa0aa167982882